src/core.jl

```diff
@@ -30,3 +30,23 @@ function pinv2x2(M::AbstractArray)
      D = SMatrix{2,2,Float64}( S[1,1] > tol ?  1/S[1,1] : 0.0 , 0.0, 0.0, S[2,2] > tol ?  1/S[2,2] : 0.0 )
      U*D*V'
 end
+
+abstract type AbstractTrackerTargetState end
+mutable struct BoostingTrackerTargetState <: AbstractTrackerTargetState
+    #Initialized
+    position::SVector{2, Float64}
+    height::Integer
+    width::Integer
+    is_target::Bool
+
+    #Uninitialized
+    responses::Array{T, 2} where T
```

Comment:
What is the purpose of `responses`? If it is needed, then you should declare `T` as part of the `struct`, for example:
`mutable struct BoostingTrackerTargetState{T <: AbstractArray} <: AbstractTrackerTargetState`
  `responses::T` 
See: https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Avoid-containers-with-abstract-type-parameters-1